### DISCUSSION TITLE
Localization of state values

### DISCUSSION THREAD

**Discussion by jensrossbach** [astrandb/miele#236] [Q&A] [answered]
Hi,

I'm using the Miele integration in combination with the custom button card for displaying information on a dashboard. Unfortunately it seems the button card cannot make use of the localization of state values as provided by the Miele integration, i.e., it always displays the raw state values. I tried to use the `hass.localize()` function in the templated output of the button card and passing the key from the translation JSON files of the Miele integration, but it seems it is not working.

I also tried using the values from the attribute "Localized" (which seems to relate to the "value_localized" properties of the Miele API). Unfortunately these values are only provided in English (so not much improvement compared to the raw state values). It seems that these values are correctly translated when providing a language parameter with API URL. Is this parameter applied in the integration and if yes, how to configure the language (German in my case)?

Are there any other hints how to get the state values translated in the button card?

**Comment by astrandb** [comment]
Tre localiization of states, progarm id etc is handled by Home Assistant. We just supply the translation files. It is a bit surprising that you get th eraw values. It indicates that the integration cannot find a mapping. Are you using built-in button card or a custom variant?  In the latter case you should contact the author of the custom card.

**Reply by jensrossbach** [reply]
Hello, thanks for the feedack. Actually I'm not using the built-in button but the [custom button card](https://github.com/custom-cards/button-card) instead. It looks like this card is not maintained anymore, so I see not much chance to get it fixed on that side.

But what about the translations that come directly from the Miele cloud (i.e., the "Localized" attributes)? I think they could be useful if they would be translated into the correct language.

**Reply by astrandb** [reply, marked as answer]
> But what about the translations that come directly from the Miele cloud (i.e., the "Localized" attributes)? I think they could be useful if they would be translated into the correct language.

Unfortunately the localization in Miele API is incomplete and sometimes absent. Therefore we chose to base the localization in the integration on the built-in features in Home Assistant. There are pros and cons either way, but now states and program names can be localized to all languages that HA supports and that is many more than Miele supports. We just need volunteers to do translation work.
